Event onReady

frenicohansen opened the pull request:
### Background

As I was working with the library, I encountered a function called onPositionComplete. According to its documentation, this function was meant to "add a listener used to execute a function once all items in the grid have been properly positioned." However, something about the name and the description felt off. It gave the impression that the listener would only be triggered a single time, likely on the first positioning of the items in the grid.

As I dug deeper into the code, it became clear that onPositionComplete actually behaved differently than what the name and documentation suggested. The listener wasn't just a one-time thing—it would trigger every time the positionItems() method was called or when the container was resized. This behavior was useful, but the existing naming and documentation didn't make that clear.

### Changes
- Introduced `onReady` to run a listener only once, after the first positioning of items.
  - This is useful for scenarios like managing a loading state, where you only want to take action once the grid is properly positioned for the first time, as mentioned in #56.
- Fixed the return type of the original onPositionComplete from void to number to match its actual behavior.
- Updated documentation for onPositionComplete and onGridReady

e-oj commented:
Nice idea! I think we can just emit a `ready` event in the listen function, add an `onReady` function, and leave everything else the same. That way there won't be any breaking changes with this feature. 

The current setup, where `positionComplete` is emitted whenever `positionItems` is called, makes sense to me so I don't think we have to add an `onRepositionComplete`

e-oj commented:
Also you're right about the documentation; we should update that to make it clearer as well. Thanks for pointing this out!

frenicohansen commented:
> Nice idea! I think we can just emit a `ready` event in the listen function, add an `onReady` function, and leave everything else the same. That way there won't be any breaking changes with this feature.
> 
> The current setup, where `positionComplete` is emitted whenever `positionItems` is called, makes sense to me so I don't think we have to add an `onRepositionComplete`

I updated the PR description and implemented your suggestion. With this there should be no breaking changes.

e-oj commented on index.d.ts:
```diff
@@ -165,11 +165,17 @@ declare class MagicGrid extends EventEmitter {
      * @param callback - function to be called on positioning complete
      */
-    onPositionComplete(callback: () => void): void;
+    onPositionComplete(callback: () => void): number;
```
Nice catch!

e-oj commented on README.md:
```diff
@@ -163,8 +163,27 @@ magicGrid.positionItems();
+
 #### .onPositionComplete(callback)
-Adds a listener used execute a function once all items in the grid have been properly positioned.
+Adds a listener that executes a function once the `positionItems()` method has been called or after the container has been resized.
```
```suggestion
Adds a listener that executes a function whenever `positionItems` is called. 
**Note: `positionItems` is called during initial setup and whenever the container or window is resized**
```

e-oj commented:
Awesome work. Thanks!
